pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java

```diff
@@ -90,7 +95,23 @@ protected void buildUpQuery(List<String> filterValues, FilterStrategy filterStra
             baseQueryBuilder = BaseQueryBuilder.of(queryPayload);
             List<QueryBuilder> queryBuilders =
                     QueryStrategyFactory.create(new QueryStrategyFactoryPayload(baseQueryBuilder, operator, parser)).build();
-            filterStrategy.build(boolQueryBuilder, baseQueryBuilder.build(queryBuilders));
+
+            if (StringUtils.isNotBlank(path)) {
+                if (!nestedQueryBuilderLists.containsKey(path)) {
+                    nestedQueryBuilderLists.put(path, queryBuilders);
+                    correctBaseQueryBuilder.put(path, baseQueryBuilder);
+                } else {
+                    List<QueryBuilder> queryBuildersFromMap = nestedQueryBuilderLists.get(path);
+                    queryBuildersFromMap.addAll(queryBuilders);
+                }
+            } else {
+                filterStrategy.build(boolQueryBuilder, baseQueryBuilder.build(queryBuilders));
+            }
+        }
+        if (!nestedQueryBuilderLists.isEmpty() && nestedQueryBuilderLists.size() == correctBaseQueryBuilder.size()) {
```

Comment:
Can the size ever not match?

Reply:
not really, but I am still paranoid

Reply:
Maybe then we can throw an error in that case to know why this part was skipped?